c_src/sasl_auth.c

```diff
@@ -41,8 +41,9 @@ typedef struct {
     int mech_set;
 } sasl_state_t;
 
-// SASL connection state
-static ErlNifResourceType* sasl_resource = NULL;
+// SASL connection nif resource
+static ErlNifResourceType* sasl_client_connection_nif_resource_type = NULL;
```

Comment:
this naming is a bit odd to me, maybe 
```suggestion
static ErlNifResourceType* sasl_client_connection_nif_resource = NULL;
```